lib/winston/create-logger.js

```diff
@@ -40,7 +40,7 @@ module.exports = function (opts) {
       // Optimize the hot-path which is the single object.
       //
       if (arguments.length === 1) {
-        const info = msg.message && msg || { message: msg };
+        const info = msg && msg.message && msg || { message: msg };
```

Comment:
I think this would be more clear if you used a ternary:
`msg && msg.message ? msg : { message: msg }`

Reply:
I found this check is not enough. The Error object can still go in, because it has a message property. The correct behavior should be:

```js
{
  message: myErrorObject,
  ...
}
```

so I made this pull request: https://github.com/winstonjs/winston/pull/1229

Reply:
Chatted about this with @DABH in our maintenance call today. The expected semantics around logging `Error` instances is that when the following are performed:

```
const err = new Error('Hey it broke yo');
logger.log('info', err);
logger.info(err);
```

the expectation will be that `err.level` is set to `'info'` and the `err` object itself becomes the `info` used in the object-stream.